### PR TITLE
feat: add tags data loading

### PR DESCRIPTION
**Pull request by azayz**

### CODE REVIEW

File: now/data_loading/data_loading.py [hunk 1]
```diff
@@ -54,6 +58,33 @@ def load_data(app: JinaNOWApp, user_input: UserInput) -> DocumentArray:
...
+            continue
+        data_folder = d.uri.split('/')[-2]
+        for tag in da:
+            if not tag.uri.endswith('.json'):
+                continue
+            tag_folder = tag.uri.split('/')[-2]
+            if tag_folder == data_folder:
+                with open(tag.uri, 'r') as f:
+                    data = json.load(f)
+                    for tag, value in data['tags'].items():
```
Comment (florian-hoenicke) [inline now/data_loading/data_loading.py:82]: Level of intents is too deep.

File: now/data_loading/data_loading.py [hunk 2]
```diff
@@ -36,6 +37,9 @@ def load_data(app: JinaNOWApp, user_input: UserInput) -> DocumentArray:
             da = _load_from_disk(app, user_input)
         elif user_input.custom_dataset_type == DatasetTypes.S3_BUCKET:
             da = _list_files_from_s3_bucket(app=app, user_input=user_input)
+            if _detect_json(da):
```
Comment (florian-hoenicke) [inline now/data_loading/data_loading.py:40]: I think, you don't need `_detect_json`.
Json is checked in the other function already.

**Review by florian-hoenicke — CHANGES_REQUESTED** [review]

File: now/data_loading/data_loading.py [hunk 3]
```diff
@@ -54,6 +57,24 @@ def load_data(app: JinaNOWApp, user_input: UserInput) -> DocumentArray:
...
+def _load_tags_from_json(da: DocumentArray):
+
+    print('Loading tags!')
+    da = sorted(da, key=lambda doc: doc.uri)
+    for i in range(0, len(da) - 1, 2):
+        if da[i + 1].uri.endswith('.json'):
+            with open(da[i + 1].uri, 'r') as f:
+                data = json.load(f)
+            for tag, value in data['tags'].items():
+                da[i].tags[tag] = value['slug']
```
Comment (makram93) [inline now/data_loading/data_loading.py:69]: What is this `slug`? Is this specific to any single use case?

Comment (azayz) [reply now/data_loading/data_loading.py:69]: yep it's taken from lottiefiles's json files for tags, but we agreed to enforce this structure on anyone who wants to add tags to their data

`{"tags":{"custom":{"name":"moneystack","slug":"moneystack"},"colors":{"name":"4A0044","slug":"4a0044"},"ml":{"name":"Slope","slug":"slope"}},"id":"c28d069f-e1af-4bd5-ab29-acddaa4dcf3d","link":"https:\/\/lottiefiles.com\/2062-moneystack","title":"Moneystack"}`

**Review by makram93 — APPROVED** [review]

File: tests/unit/data_loading/test_data_loading.py [hunk 4]
```diff
@@ -123,3 +123,14 @@ def test_da_custom_ds(da: DocumentArray):
...
+
+def test_load_tags(gif_resource_path: str):
+    da = DocumentArray.from_files(gif_resource_path)
+
+    da = _load_tags_from_json(da)
+
+    for d in da:
+        if d.uri.endswith('.gif'):
+            assert 'custom' in d.tags
+            assert 'ml' in d.tags
```
Comment (florian-hoenicke) [inline tests/unit/data_loading/test_data_loading.py:136]: assert values

File: now/data_loading/data_loading.py [hunk 5]
```diff
@@ -54,6 +57,24 @@ def load_data(app: JinaNOWApp, user_input: UserInput) -> DocumentArray:
     return da
 
 
+def _load_tags_from_json(da: DocumentArray):
+
+    print('Loading tags!')
+    da = sorted(da, key=lambda doc: doc.uri)
```
Comment (florian-hoenicke) [inline now/data_loading/data_loading.py:63]: 1. build dictionary (path->(file1.gif, file2.json))
2. merge files to one document if content file exists

**Review by florian-hoenicke — CHANGES_REQUESTED** [review]

**Pull request closed without merging**